robotools/evotools/__init__.py

```diff
@@ -774,6 +1499,168 @@ def dispense(
+            one of 'auto' (default), 'source' or 'destination'
+                'auto': partitioning by source unless the source is a Trough
+                'source': partitioning by source columns
+                'destination': partitioning by destination columns
+        kwargs
+            Additional keyword arguments to pass to aspirate and dispense.
+            Take a look at `Worklist.aspirate_well` for the full list of options.
+        """
+        # reformat the convenience parameters
+        source_wells = numpy.array(source_wells).flatten("F")
+        destination_wells = numpy.array(destination_wells).flatten("F")
+        volumes = numpy.array(volumes).flatten("F")
+        nmax = max((len(source_wells), len(destination_wells), len(volumes)))
+
+        if len(source_wells) == 1:
+            source_wells = numpy.repeat(source_wells, nmax)
+        if len(destination_wells) == 1:
+            destination_wells = numpy.repeat(destination_wells, nmax)
+        if len(volumes) == 1:
+            volumes = numpy.repeat(volumes, nmax)
+        lengths = (len(source_wells), len(destination_wells), len(volumes))
+        assert (
+            len(set(lengths)) == 1
+        ), f"Number of source/destination/volumes must be equal. They were {lengths}"
+
+        # automatic partitioning
+        partition_by = _optimize_partition_by(source, destination, partition_by, label)
+
+        # the label applies to the entire transfer operation and is not logged at individual aspirate/dispense steps
+        self.comment(label)
+        nsteps = 0
+        lvh_extra = 0
+
+        for srcs, dsts, vols in _partition_by_column(source_wells, destination_wells, volumes, partition_by):
+            # make vector of volumes into vector of volume-lists
+            vols = [
+                _partition_volume(float(v), max_volume=self.max_volume) if self.auto_split else [v]
+                for v in vols
+            ]
+            # transfer from this source column until all wells are done
+            npartitions = max(map(len, vols))
+            # Count only the extra steps created by LVH
+            lvh_extra += sum([len(vs) - 1 for vs in vols])
+            for p in range(npartitions):
+                naccessed = 0
+                # iterate the rows
+                for s, d, vs in zip(srcs, dsts, vols):
+                    # transfer the next volume-fraction for this well
+                    if len(vs) > p:
+                        v = vs[p]
+                        if v > 0:
+                            self.evo_aspirate(source, s, source_position, v, liquid_class, tips, label=None)
+                            self.evo_dispense(
+                                destination, d, destination_position, v, liquid_class, tips, label=None
+                            )
+                            nsteps += 1
+                            # if wash_scheme is not None:
+                            #     self.wash(scheme=wash_scheme)
```

Comment:
ToDo ?

Reply:
I did not plan to stage, let alone push the evo_transfer() method yet so I deleted it for the time being. After the rest works and the method has been tested, I will add it again (since it is mostly for convenience).